chain/chunks/src/lib.rs

```diff
@@ -1075,6 +1092,142 @@ impl ShardsManager {
         )
     }
 
+    /// Constructs a vector of receipt proofs for specified chunk.
+    ///
+    /// The `make_elem` argument converts `ReceiptsProof` values into objects
+    /// which are returned from this function.  In practice this is used to
+    /// either return `ReceiptProof` directly or wrap the object into `Arc`.
+    fn make_outgoing_receipts_proofs<T>(
+        &self,
+        chunk_header: &ShardChunkHeader,
+        outgoing_receipts: &[Receipt],
+        make_elem: impl Fn(ReceiptProof) -> T,
+    ) -> Result<Vec<T>, near_chunks_primitives::Error> {
+        let shard_id = chunk_header.shard_id();
+        let shard_layout = self
+            .runtime_adapter
+            .get_shard_layout_from_prev_block(chunk_header.prev_block_hash_ref())?;
+
+        let hashes = Chain::build_receipts_hashes(&outgoing_receipts, &shard_layout);
+        let (root, proofs) = merklize(&hashes);
+        if chunk_header.outgoing_receipts_root() != root {
+            return Err(near_chunks_primitives::Error::UnexpectedOutgoingRemeiptsRoot {
```

Comment:
I think we should assert here like the previously instead of returning an error. This means we have accepted an invalid chunk in the storage and there is something seriously wrong.

Reply:
Works for me. Makes the code simpler.